base/deprecated.jl

```diff
@@ -1348,6 +1347,44 @@ end
 @deprecate srand(filename::AbstractString, n::Integer=4) srand(read!(filename, Array{UInt32}(Int(n))))
 @deprecate MersenneTwister(filename::AbstractString)  srand(MersenneTwister(0), read!(filename, Array{UInt32}(Int(4))))
 
+# also remove all support machinery in src for current_module when removing this deprecation
+_current_module() = ccall(:jl_get_current_module, Ref{Module}, ())
+@noinline function binding_module(s::Symbol)
+    depwarn("binding_module(symbol) is deprecated, use `binding_module(__module__, symbol)` instead.", :binding_module)
+    return binding_module(_current_module(), s)
+end
+@noinline function expand(x::ANY)
+    depwarn("expand(x) is deprecated, use `expand(__module__, x)` instead.", :expand)
```

Comment:
For the REPL, you can just type `Main`. There's nothing particularly special about the name `__module__` that makes it any different from `module`. I thought it was nicely reminiscent of the macro argument name (allowing easier search), and also provided underlining of the change (highlighting word diff).